shark-graph/src/test/java/shark/BytesMapTest.kt

```diff
@@ -0,0 +1,108 @@
+package shark
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import shark.internal.UnsortedBytesMap
+
+class BytesMapTest {
+
+  @Test fun writeReadLongValue() {
+    val unsortedMap = UnsortedBytesMap(bytesPerValue = 8, longIdentifiers = false)
+    unsortedMap.append(1)
+        .apply {
+          writeLong(Long.MIN_VALUE)
+        }
+
+    val array = unsortedMap.moveToSortedMap()[1]!!
+    assertThat(array.readLong()).isEqualTo(Long.MIN_VALUE)
+  }
+
+  @Test fun writeReadTruncatedLongValue() {
+    val maxUnsigned3Bytes = 0x00000FFFL
+    val unsortedMap = UnsortedBytesMap(bytesPerValue = 3, longIdentifiers = false)
+    unsortedMap.append(1)
+        .apply {
+          writeTruncatedLong(maxUnsigned3Bytes, 3)
+        }
+
+    val array = unsortedMap.moveToSortedMap()[1]!!
+    assertThat(array.readTruncatedLong(3)).isEqualTo(maxUnsigned3Bytes)
+  }
+
+  @Test fun bar() {
+    val unsortedMap = UnsortedBytesMap(bytesPerValue = 1, longIdentifiers = true)
+    unsortedMap.append(42)
+        .apply {
+          writeByte(4)
+        }
+    unsortedMap.append(0)
+        .apply {
+          writeByte(3)
+        }
+    unsortedMap.append(3)
+        .apply {
+          writeByte(20)
+        }
+    unsortedMap.append(Long.MAX_VALUE)
+        .apply {
+          writeByte(127)
+        }
+    val sortedEntries = unsortedMap.moveToSortedMap()
+        .entrySequence()
+        .toList()
+
+    assertThat(sortedEntries.map { it.first }).containsExactly(0, 3, 42, Long.MAX_VALUE)
+    assertThat(
+        sortedEntries.map {
+          byteArrayOf(
+              it.second.readByte()
+          )
+        }).containsExactly(
+        byteArrayOf(3), byteArrayOf(20), byteArrayOf(4),
+        byteArrayOf(127)
+    )
+  }
+
+  @Test fun foo() {
```

Comment:
Are `foo` and `bar` test names temporary here?

Reply:
Great test names isn't it?

I'm about to push an update that changes this.